gui/wxpython/gui_core/widgets.py

```diff
@@ -864,6 +864,52 @@ def _mapNameValidationFailed(ctrl):
                                   _mapNameValidationFailed)
 
 
+class GenericMultiValidator(Validator):
+    """This validator checks conditions and calls callbacks
+    in case the condition is not fulfilled.
+    """
+
+    def __init__(self, checks):
+        """Standard constructor.
+
+        :param checks: list of tuples consisting of conditions (list of
+        functions which accepts string value and returns T/F) and callbacks (
+        list of functions which is called when condition is not fulfilled)
+        """
+        Validator.__init__(self)
+        self._checks = checks
+
+    def Clone(self):
+        """Standard cloner.
+
+        Note that every validator must implement the Clone() method.
+        """
+        return GenericMultiValidator(self._checks)
+
+    def Validate(self, win):
```

Comment:
This is not correct, please try to run the code line by line in your mind or on paper and see what the problem is.